server/src/test/java/org/opensearch/index/shard/ReplicaRecoveryWithRemoteTranslogOnPrimaryTests.java

```diff
@@ -58,6 +61,8 @@ public void testStartSequenceForReplicaRecovery() throws Exception {
             final IndexMetadata newIndexMetadata = IndexMetadata.builder(replica.indexSettings().getIndexMetadata())
                 .primaryTerm(replicaRouting.shardId().id(), replica.getOperationPrimaryTerm() + 1)
                 .build();
+            Directory storeDirectory = ((FilterDirectory) ((FilterDirectory) replica.store().directory()).getDelegate()).getDelegate();
+            ((BaseDirectoryWrapper) storeDirectory).setCheckIndexOnClose(false);
```

Comment:
Why are we preventing checks on directory close ? I see these checks are important as they validate store for readable commits before close and are enabled by default. This applies to all changes in other files as well.

Reply:
Same reason as above.